### PR TITLE
framework/media : Correct wrong casting type.

### PR DESCRIPTION
**Pull request by dev-hue**
- SocketOutputDataSource::write return type is ssize_t

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [dd3f0810c28247e93ee3271e783e9fe9b6a1831f] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [dd3f0810c28247e93ee3271e783e9fe9b6a1831f] - Code Rule Check (C++) OK.

File: framework/src/media/SocketOutputDataSource.cpp [hunk 1]
```diff
@@ -100,7 +100,7 @@ bool SocketOutputDataSource::isPrepare()
 ssize_t SocketOutputDataSource::write(unsigned char* buf, size_t size)
 {
 	if (!buf) {
-		return (size_t)0;
+		return (ssize_t)0;
```
Comment (sunghan-chang) [inline framework/src/media/SocketOutputDataSource.cpp]: Is typecasting needed in this case?

Comment (Hyunjun85) [reply framework/src/media/SocketOutputDataSource.cpp]: typedef uint16_t size_t;
typedef int16_t ssize_t;

It looks like to execute a downcasting.
Isn't there any possibility about data missing?

Comment (JeongJunSik) [reply framework/src/media/SocketOutputDataSource.cpp]: i think

"return EOF;"
This seems to be right.

Comment (dev-hue) [reply framework/src/media/SocketOutputDataSource.cpp]: I think so "return EOF" and typecasting does not seem necessary.

How about FileOutputDataSource::write and BufferOutputDataSource::write?
FileOutputDataSource::write returns `(ssize_t)EOF`
BufferOutputDataSource::write returns `EOF`

Comment (Taejun-Kwon) [reply framework/src/media/SocketOutputDataSource.cpp]: I think type casting is not neccessary, EOF is -1.
Also those of 3 datasources should return EOF.
(Of course we have to define some error types in future)

**Comment by gcjjyy** [comment]
size_t is just a parameter's type. Return value doesn't need to be same type with a parameter's type. Isn't it? About data loss, the function has to control it. This is not a type casting problem.

File: framework/src/media/SocketOutputDataSource.cpp (hunk 1 above)
Comment (gcjjyy) [reply framework/src/media/SocketOutputDataSource.cpp]: size_t is just a parameter's type. Return value doesn't need to be same type with a parameter's type. About data loss, The function logic has to control it. This is not a type casting problem I think.

**Comment by seinfra** [comment]
Target : [b1e68108f972dd54f282148affa701311808a8a8] - Code Rule Check OK.

**Pull request merged**